crates/rust-analyzer/src/config.rs

```diff
@@ -434,6 +434,65 @@ config_data! {
         completion_autoself_enable: bool        = true,
         /// Whether to add parenthesis and argument snippets when completing function.
         completion_callable_snippets: CallableCompletionDef  = CallableCompletionDef::FillArguments,
+        /// A list of full paths to traits to exclude from flyimport.
+        ///
+        /// Traits in this list won't be suggested to be imported by flyimport for their methods. Methods from them won't be available in flyimport completion. They will still be available if in scope.
+        ///
+        /// Note that the trait themselves can still be suggested by flyimport.
+        ///
+        /// This setting also inherits `#rust-analyzer.completion.excludeTraits#`.
+        completion_excludeFlyimportTraits: Vec<String> = Vec::new(),
+        /// A list of full paths to traits to exclude from completion.
+        ///
+        /// Methods from these traits won't be completed, even if the trait is in scope. However, they will still be suggested on expressions whose their type is `dyn Trait`, `impl Trait` or `T where T: Trait`.
+        ///
+        /// Note that the trait themselves can still be completed.
+        ///
+        /// This setting defaults to:
+        ///
+        ///  - [`core::convert::AsRef`](https://doc.rust-lang.org/nightly/core/convert/trait.AsRef.html)
+        ///  - [`core::convert::AsMut`](https://doc.rust-lang.org/nightly/core/convert/trait.AsMut.html)
+        ///  - [`core::borrow::Borrow`](https://doc.rust-lang.org/nightly/core/borrow/trait.Borrow.html)
+        ///  - [`core::borrow::BorrowMut`](https://doc.rust-lang.org/nightly/core/borrow/trait.BorrowMut.html)
+        ///  - [`core::cmp::PartialEq`](https://doc.rust-lang.org/nightly/core/cmp/trait.PartialEq.html)
+        ///  - [`core::any::Any`](https://doc.rust-lang.org/nightly/core/any/trait.Any.html)
+        ///  - All operator traits (in [`core::ops`](https://doc.rust-lang.org/nightly/core/ops))
+        ///
+        /// Note that if you override this setting, those traits won't be automatically inserted, so you may want to insert them manually.
```

Comment:
```suggestion
        /// Note that if this setting is overridden, the above traits must be re-inserted manually override.
```

Reply:
This implies that the user needs to insert those traits, while the reality is that we don't include them when overridden so they can be removed if wanted.